### DISCUSSION TITLE
'Earlybird' Bonus for Beta Katas

### DISCUSSION THREAD

**Discussion by Kacarott** [codewars/codewars.com#2389] [Ideas]
The beta process for more difficult katas (4kyu+) can be quite slow, and many quality katas will only get 0-2 solves before being 'forgotten' in the depths that is beta. My proposal is to add an "Earlybird" bonus for the first few solvers of a kata.

Currently, solving a beta kata earns the solver `4` points. When/if that beta kata is approved, they are given the rest of the points they are 'owed' as if they had solved the kata in its approved state. The Earlybird bonus would simply be a multiplier (or some similar function) of these 'owed' points. 

For example, perhaps the first three solvers (author not included) of a brand new beta kata earn a `1.1` multiplier, and the next three solvers earn a `1.05` multiplier. Usually, 6 solves is enough to bring a kata to an approvable state (assuming it is a quality kata) so further solves would not get a bonus. All the solvers would still only get the initial `4` honour, however when the kata is approved, the 'Earlybird' bonus would be applied (For example, the first solver of a newly approved Purple kata might earn `140` honour instead of the normal `128`).

I believe this would encourage more people to solve beta katas, in particular harder katas which are almost always under 6 solves, and would provide a larger 'bounty' when approved. I also think the point bonus is a fair 'compensation' for the risk in solving a beta kata which might never be approved. Overall I think it would help speed up the beta process, and see many more quality (and more difficult) katas being approved.

Let me know what you think!

**Comment by hobovsky** [comment]
I generally like the idea, but I see a couple of potential issues:

- Getting a (potentially very rewarding) bounty would require little effort of, for example, hardcoding a couple of inputs. Early versions of kata are expected to have weak tests, what would mean that the biggest reward is granted for the smallest effort.
- Reward does not mean that kata will end up better. Users could (would?) jump on a new beta, solve it, grab the reward, and downvote it into oblivion, or upvote albeit of being bad, or just leave no feedback.

**Reply by Kacarott** [reply]
I did consider the first point, and I think there would have to be an additional detail that invalidated solutions do not qualify for earlybird bonus. (which is then part of the 'risk' which the bonus itself is rewarding).

As for the second point, the reward would not be given out until the beta kata is approved (if ever). This might cause people to vote higher satisfaction (so that they can earn their reward), however this incentive already exists in the beta process (the early bird bonus is just a bit more) and I don't think too many people abuse the satisfaction rating merely for points. (I could be wrong).
